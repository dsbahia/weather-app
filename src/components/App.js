import "../styles/App.css";
import React, { useState, useEffect } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import SearchForm from "./searchForm";

function App() {
  const [location, setLocation] = useState({
    city: "",
    country: "",
  });

  const [forecasts, setForecasts] = useState([]);
  const [selectedDate, setSelectedDate] = useState(0);
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation, setErrorMessage);
  }, []);

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate,
  );

  const [searchText, setSearchText] = useState("");

  const handleCitySearch = () => {
    getForecast(
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage,
      searchText,
    );
  };

  return (
    <div className="weather-app">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;
