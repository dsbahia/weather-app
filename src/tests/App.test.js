import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders App component correctly", () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    render(<App />);
    const h1Element = screen.getByText(/Manchester, UK/i);
    expect(h1Element).toBeInTheDocument();
  });
});
