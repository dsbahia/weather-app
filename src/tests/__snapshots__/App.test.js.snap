// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <div />
    <div
      class="search-form"
    >
      <input
        class="search-form"
        type="text"
        value=""
      />
      <button
        class="search-form"
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
